### PR TITLE
Fix composer supported PHP versions

### PR DESCRIPTION
**Pull request by KadelaKuba**
### WHY are these changes introduced?
This project has wrong supported php versions. The ^8.1 notation in composer means that the supported PHP version is >=8.1.0 < 9.0.0-0. But this project is not ready for PHP 8.2.

### WHAT is this pull request doing?
This pull request updates the supported PHP versions in composer.json. 

There are three types of PHP verions supported now:

- ^7.4 (>=7.4.0 <8.0.0-0)
- ^8.0 (>=8.0.0 <8.1.0-0)
- ^8.1.0 (>=8.1.0 <8.2.0-0)

## Type of change

- [x] Patch: Bug (non-breaking change which fixes an issue)
- [ ] Minor: New feature (non-breaking change which adds functionality)
- [ ] Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have added a changelog entry, prefixed by the type of change noted above
- [x] I have added/updated tests for this change
- [x] I have updated the documentation for public APIs from the library (if applicable)

### CODE REVIEW

**Comment by KadelaKuba** [comment]
I've signed the CLA!"

**Comment by KadelaKuba** [comment]
I've signed the CLA!

**Comment by KadelaKuba** [comment]
I've signed the CLA!

File: composer.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
         }
     },
     "require": {
-        "php": "^7.4 || ^8.0 || ^8.1",
+        "php": "^7.4 || ~8.0 || ~8.1.0",
```
Comment (fredden) [inline composer.json:34]: `~1.0` is equivalent to `^1.0` or `1.*`, but `~1.0.0` is equivalent to `1.0.*`.

```suggestion
        "php": "^7.4 || ~8.0.0 || ~8.1.0",
```

**Comment by Jan0707** [comment]
Closing this in favor of https://github.com/Shopify/shopify-api-php/pull/264 as this would otherwise require updates for the 7.4 version.

**Pull request closed without merging**